src/main/java/com/xceptance/xlt/util/TestCaseMapper.java

```diff
@@ -0,0 +1,265 @@
+     * @param testCaseName
+     *            the test case name
+     * @param testClassNames
+     *            the set of test class names
+     * @return the single test class name matching the test case name
+     * @throws XltException
+     *             if no or multiple class names in the set match the given test case name
+     */
+    private String getSingleMatchingTestClassName(final String testCaseName, final Set<String> testClassNames) throws XltException
+    {
+        final Pattern testClassNamePattern = Pattern.compile(getTestClassNamePattern(testCaseName));
+        final List<String> matchingClassNames = testClassNames.stream().filter(testClassNamePattern.asPredicate())
+                                                              .collect(Collectors.toList());
+
+        if (matchingClassNames.isEmpty())
+        {
+            throw new XltException("Auto-mapping test cases failed because no matching test class was found for test case '" +
+                                   testCaseName + "'.");
+        }
+
+        if (matchingClassNames.size() > 1)
+        {
+            throw new XltException("Auto-mapping test cases failed because multiple matching test classes were found for test case '" +
+                                   testCaseName + "': [ " + StringUtils.join(", ", matchingClassNames, 5, ", ...") + " ].");
+        }
+
+        return matchingClassNames.get(0);
+    }
+
+    /**
+     * Get the RegEx pattern for class files that match any of the given test case names.
+     *
+     * @param testCaseNames
+     *            the test case names
+     * @return the RegEx pattern
+     */
+    static String getTestClassesFilePattern(final Set<String> testCaseNames)
+    {
+        final Set<String> escapedTestCaseNames = testCaseNames.stream().map(Pattern::quote).collect(Collectors.toSet());
+        return "(^|(.*(\\/|\\$)))(" + String.join("|", escapedTestCaseNames) + ")\\.class$";
+    }
+
+    /**
+     * Get the RegEx pattern for class names that match the given test case name.
+     *
+     * @param testCaseName
+     *            the test case name
+     * @return the RegEx pattern
+     */
+    static String getTestClassNamePattern(final String testCaseName)
+    {
+        return "(^|(.*(\\.|\\$)))" + Pattern.quote(testCaseName) + "$";
+    }
+
+    /**
+     * Implementation of {@link Scanner} to scan for all classes in a directory.
+     */
+    private class AllClassesScanner implements Scanner, QueryBuilder
+    {
+        @Override
+        public List<Map.Entry<String, String>> scan(ClassFile classFile)
+        {
+            // classes are stored with a mapping to their parent classes using the Reflections library
+            // since we want all classes regardless of parent we set the parent to "Object.class"
+            final List<Map.Entry<String, String>> entries = new ArrayList<>();
+            entries.add(entry(Object.class.getName(), classFile.getName()));
+            return entries;
```

Comment:
Tiny optimization for single-entry lists:

```suggestion
            return List.of(entry(Object.class.getName(), classFile.getName()));
```